pkg/yurthub/server/nonresourceproxy.go

```diff
@@ -0,0 +1,99 @@
+/*
+Copyright 2020 The OpenYurt Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package server
+
+import (
+	"context"
+	"github.com/gorilla/mux"
+	"github.com/openyurtio/openyurt/cmd/yurthub/app/config"
+	"github.com/openyurtio/openyurt/pkg/yurthub/kubernetes/rest"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/klog/v2"
+	"net/http"
+)
+
+var nonResourceReqPaths = []string{
+	"/version",
+	"/apis/discovery.k8s.io/v1",
+	"/apis/discovery.k8s.io/v1beta1",
+}
+
+var cfg *config.YurtHubConfiguration
+var rcm *rest.RestConfigManager
+
+func wrapNonResourceHandler(proxyHandler http.Handler, config *config.YurtHubConfiguration, restConfigMgr *rest.RestConfigManager) http.Handler {
```

Comment:
I think it's better to implement the new handler at `pkg/yurthub/proxy` instead of here, because it's somewhere the responsibility of yurtReverseProxy.

Reply:
And do we have better name for this func? It does not only add the NonResourceHandler but also ResourceHandler, it's confused calling it as warpNonResourceHandler.

Reply:
I think there may be more non-resource requests should be handled  and cached in the future and this file is to wrap the yurt-reverse-proxy and intercept the non-resource requests. I want to consult @rambohe-ch for his opinions.

Reply:
@Sodawyx @Congrool Maybe it's ok to locate handler file under pkg/server, but I think no resource handler is not a proxy feature, so how about rename `nonresourceproxy.go` to `nonresource.go`?

Reply:
I think it is OK.